gadgets/trace_mount/program.bpf.c

```diff
@@ -0,0 +1,197 @@
+/* SPDX-License-Identifier: GPL-2.0 */
+/* Copyright (c) 2021 Hengqi Chen */
+#include <vmlinux.h>
+#include <bpf/bpf_helpers.h>
+#include <bpf/bpf_core_read.h>
+#include <gadget/macros.h>
+#include <gadget/mntns_filter.h>
+#include <gadget/types.h>
+
+#define MAX_ENTRIES 10240
+#define TASK_COMM_LEN 16
+#define FS_NAME_LEN 8
+#define DATA_LEN 512
+#define PATH_MAX 4096
+
+enum op {
+	MOUNT,
+	UMOUNT,
+};
+
+struct arg {
+	__u64 ts;
+	__u64 flags;
+	const char *src;
+	const char *dest;
+	const char *fs;
+	const char *data;
+	enum op op;
+};
+
+struct event {
+	__u64 delta;
+	__u64 flags;
+	__u32 pid;
+	__u32 tid;
+	__u64 mount_ns_id;
```

Comment:
This has to use the `gadget_mntns_id` type, otherwise the container enrichment won't work.